### ISSUE TITLE
Precomputed `logprobs` and `ref_logprobs` are not used by PPOOrchestrator for causal models

### ISSUE THREAD

**Issue by ZHAOTING** [CarperAI/trlx#248]
### 🐛 Describe the bug

In PPOOrchestrator's `make_experience()` function, `logprobs` and `ref_logprobs` computed from logits by the code from [L180](https://github.com/CarperAI/trlx/blob/main/trlx/orchestrator/ppo_orchestrator.py#L180) to [L189](https://github.com/CarperAI/trlx/blob/main/trlx/orchestrator/ppo_orchestrator.py#L188). 

For seq2seq models, they are later used for computing `all_logprobs` and `rewards` at L199-201. 
But for causal models, `logprobs` and `ref_logprobs` are computed again (with a correct token shift) at L210-211.

It seems that the code from L187 to L195 is unnecessary for causal models.

~~~
if self.trainer.config.model.model_arch_type == "seq2seq":
    logprobs = logprobs_from_logits(
        logits[:, :-1, :], response_tensors[:, 1:]
    )
    ref_logprobs = logprobs_from_logits(
        ref_logits[:, :-1, :], response_tensors[:, 1:]
    )
else:
    logprobs = logprobs_from_logits(logits, all_tokens)
    ref_logprobs = logprobs_from_logits(ref_logits, all_tokens)

n = samples.shape[0]
logprobs = logprobs.cpu()
ref_logprobs = ref_logprobs.cpu()
query_tensors = query_tensors.cpu()
response_tensors = response_tensors.cpu()
if self.trainer.config.model.model_arch_type == "seq2seq":
    start = 1  # skip the <s> token
    ends = (response_tensors[:, start:] != 0).sum(1)
    all_logprobs = [logprobs[ix, start : ends[ix]] for ix in range(n)]
    all_values = [values[ix, start - 1 : ends[ix] - 1] for ix in range(n)]
    rewards = [
        -self.trainer.kl_ctl.value
        * (
            logprobs[ix, start : ends[ix]]
            - ref_logprobs[ix, start : ends[ix]]
        )
        for ix in range(n)
    ]
else:
    logprobs = logprobs_from_logits(logits[:, :-1, :], all_tokens[:, 1:])
    ref_logprobs = logprobs_from_logits(
        ref_logits[:, :-1, :], all_tokens[:, 1:]
    )

    n = samples.shape[0]
    values = values.cpu()[:, :-1]
    logprobs = logprobs.cpu()
    ref_logprobs = ref_logprobs.cpu()
    query_tensors = query_tensors.cpu()
    response_tensors = response_tensors.cpu()

    start = query_tensors.shape[1] - 1
    ends = start + attention_mask[:, start:].sum(1)
    all_values = [values[ix, start : ends[ix]] for ix in range(n)]
    all_logprobs = [logprobs[ix, start : ends[ix]] for ix in range(n)]

    rewards = -self.trainer.kl_ctl.value * (logprobs - ref_logprobs)
    rewards = [rs[start : ends[ix]] for ix, rs in enumerate(rewards)]
~~~

### Which trlX version are you using?

The lastest commit at 7fe0a50d0250bd1bf540bd949537cf775f42241a

**Comment by reciprocated** [comment]
Hi, you are correct, this was cleaned up in https://github.com/CarperAI/trlx/commit/9745ec360ee43cafd3b63f213dbf04ca212e5e35 and will be merged shortly

**Comment by ZHAOTING** [comment]
Thanks!
